mlir/lib/Dialect/MQTOpt/Transforms/MergeRotationGatesPattern.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "mlir/Dialect/MQTOpt/IR/MQTOptDialect.h"
+#include "mlir/Dialect/MQTOpt/Transforms/Passes.h"
+#include "mlir/IR/BuiltinAttributes.h"
+
+#include <algorithm>
+#include <cstddef>
+#include <iterator>
+#include <mlir/Dialect/Arith/IR/Arith.h>
+#include <mlir/IR/MLIRContext.h>
+#include <mlir/IR/Operation.h>
+#include <mlir/IR/PatternMatch.h>
+#include <mlir/IR/Value.h>
+#include <mlir/IR/ValueRange.h>
+#include <mlir/Support/LLVM.h>
+#include <mlir/Support/LogicalResult.h>
+#include <stdexcept>
+#include <string>
+#include <unordered_set>
+
+namespace mqt::ir::opt {
+
+static const std::unordered_set<std::string> MERGEABLE_GATES = {
+    "gphase", "rx", "ry", "rz", "rxx", "ryy", "rzz", "rzx"};
```

Comment:
this is missing the `p` (phase) gate, which is the same as the `rz` gate up to global phase. Adding that might require a couple of docs and test updates throughout this PR.